src/player.cpp

```diff
@@ -6967,8 +6973,20 @@ void player::hardcoded_effects(effect &it)
     } else if (id == "grabbed") {
         blocks_left -= 1;
         dodges_left = 0;
-        // Set ourselves up for removal
-        it.set_duration(0);
+        int zed_number = 0;
+        for (int i = -1; i <= 1; i++) {
```

Comment:
You could use `for( auto &dest : g->m.points_in_radius( p, 1, 0 ) )` instead of those 2 `for`s and tripoint declaration.
You can see examples of this in src/damage.cpp.